linera-base/src/prometheus_util.rs

```diff
@@ -38,6 +38,42 @@ pub fn register_histogram_vec(
     register_histogram_vec!(histogram_opts, label_names).expect("Histogram can be created")
 }
 
+/// Construct the latencies starting from 0.0001 and ending at the maximum latency
+pub fn bucket_latencies(max_latency: f64) -> Option<Vec<f64>> {
+    let mut latencies = Vec::new();
+    let mut latency = 0.0001_f64;
+    let max_delta = 0.01;
+    loop {
+        for mult in [1.0_f64, 2.5_f64, 5.0_f64] {
+            let target_latency = latency * mult;
+            latencies.push(target_latency);
+            let delta = (target_latency - max_latency).abs() / max_latency;
+            if delta < max_delta {
+                return Some(latencies);
+            }
+        }
+        latency *= 10_f64;
+    }
+}
+
+/// Construct the bucket interval starting from a value and an ending value.
+pub fn bucket_interval(start_value: f64, end_value: f64) -> Option<Vec<f64>> {
+    let mut values = Vec::new();
+    let mut value = start_value;
+    let max_delta = 0.01;
+    loop {
+        for mult in [1.0_f64, 2.5_f64, 5.0_f64] {
+            let target_value = value * mult;
+            values.push(target_value);
+            let delta = (target_value - end_value).abs() / end_value;
+            if delta < max_delta {
+                return Some(values);
+            }
+        }
+        value *= 10_f64;
+    }
+}
+
```

Comment:
Prometheus already provides utility functions to generate these buckets, like I mentioned in the issue: https://github.com/linera-io/linera-protocol/issues/2795#issuecomment-2453014948
Couldn't we use those instead of implementing it from scratch?

Reply:
Ok, done.